### PR TITLE
Filters issue fix

### LINKED ISSUE #121: Don't have possibility to update Filters widget after data came

### PR DESCRIPTION
**Pull request by react-developer-ssa-group**
closes #121

### CODE REVIEW

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 4       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 13  | ❗       |
    


*🟡 storybook/components-multiple-dropdown--dynamically-changed-items__[w1440px]*: 


| Image |
| --- |
|  |



*🟡 storybook/components-multiple-dropdown--dynamically-changed-items__[w1920px]*: 


| Image |
| --- |
|  |



*🟡 storybook/components-multiple-dropdown--dynamically-changed-items__[w390px]*: 


| Image |
| --- |
|  |



*🟡 storybook/components-multiple-dropdown--dynamically-changed-items__[w900px]*: 


| Image |
| --- |
|  |



<details>
<summary>❗ storybook/components-multiple-dropdown--basic__[w1440px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/components-multiple-dropdown--basic__[w1920px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/components-multiple-dropdown--basic__[w390px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/components-multiple-dropdown--basic__[w900px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/components-multiple-dropdown--custom__[w1440px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/components-multiple-dropdown--custom__[w1920px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/components-multiple-dropdown--custom__[w390px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/components-multiple-dropdown--custom__[w900px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/components-multiple-dropdown--opened__[w1440px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/components-multiple-dropdown--opened__[w1920px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/components-multiple-dropdown--opened__[w390px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/components-multiple-dropdown--opened__[w900px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/utility-responsive-image--responsive-image__[w1920px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 6  | ❗       |
    


<details>
<summary>❗ custom/widgets-filters--default__[w1920px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ custom/widgets-user-profile--is-opened__[w1920px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/widgets-filters--default__[w1440px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/widgets-filters--default__[w1920px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/widgets-filters--default__[w390px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/widgets-user-profile--default__[w1440px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>

File: packages/widgets/src/components/TableFilters/hooks/useTableData.ts [hunk 1]
```diff
@@ -42,18 +44,27 @@ export const useTableData = ({
     }
   };
 
-  useEffect(() => {
-    if (initialState) {
-      const data = JSON.parse(JSON.stringify(initialState));
-      Object.keys(initialState).forEach((groupName) => {
-        const groupInfo = propOr({}, groupName)(initialState);
+  const proceedSettingCheckboxData = (newCheckboxData?: TableFilterConfig) => {
+    console.log('>>>proceedSettingCheckboxData', newCheckboxData);
```
Comment (arthur-kashaiev-ssa-group) [inline packages/widgets/src/components/TableFilters/hooks/useTableData.ts:48]: Remove `console.log()`

**Review by arthur-kashaiev-ssa-group — CHANGES_REQUESTED** [review]

File: packages/widgets/src/components/TableFilters/hooks/useTableData.ts (hunk 1 above)
Comment (react-developer-ssa-group) [reply packages/widgets/src/components/TableFilters/hooks/useTableData.ts]: Done

**Review by arthur-kashaiev-ssa-group — APPROVED** [review]

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 8  | ❗       |
    


<details>
<summary>❗ custom/widgets-filters--default__[w1920px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ custom/widgets-meal-nutrients--hint-opened__[w390px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ custom/widgets-user-profile--is-opened__[w1920px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/widgets-calories--default__[w1440px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/widgets-filters--default__[w1440px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/widgets-filters--default__[w1920px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/widgets-filters--default__[w390px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/widgets-user-profile--default__[w1920px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>

**Pull request merged**